test/index.js

```diff
@@ -259,3 +260,28 @@ test('return empty 304 response when If-None-Match matches ETag', async t => {
   t.is(statusCode, 304)
   t.is(body, '')
 })
+
+test('return HIT after empty 304 response', async t => {
```

Comment:
Not sure if this test checks what it should. I think that when you'll remove the implementation, this test will still pass.

Reply:
Actually the test would fail before the implementation. I have already tested with both `master` branch and this `master-fix-response-304-cache` branch.

Scenario before implementation (performance issue):
1. Browser send request with header `if-none-match`
2. Cache expired and server does SSR
3. Etag value is the same and content not modified
4. **Server response 304 with empty body and generated content is not saved in cache store**
* This behaviour will cause the server does SSR on subsequent request because cached data is not saved in cache store and cacheable-response is supposed to save the cached data in cache store before sending 304 response

Scenario after implementation:
1. Browser send request with header `if-none-match`
2. Cache expired and server does SSR
3. Etag value is the same and content not modified
4. **Generated content is saved to cache store before response 304 with empty body**
* Now the SSR generated content is saved in cache store and any subsequent request will use the cached data instead of generating the content again before it expires.

What the test does?
1. First request to generate cached data
2. Clear the cache store (simulate cache expiry)
3. Send second request with `if-none-match` and Etag value from first request, server would response 304 not modified
4. Send third request without header `if-none-match` and server should return cached data with `HIT` cache status (server response `MISS` status before implementation which would cause the test case fails)

I found this problem in my Next.js project in production environment because our content is mostly static with the same Etag and I found that cacheable-response didn't save the SSR generated content and caused it to execute SSR more frequently than expected.

Reply:
Okay, that makes sense. 

But the problem that I see here is that the test doesn't specifically test what you're trying to accomplish (which is not calling the `get` method when the response is 304). Returning `HIT` after empty 304 response doesn't indicate that your server would not rerender in this case. Does that make sense to you?

Reply:
What I'd propose is to create a test spy as the `get` method and check if it isn't invoked on a 304 response.

Reply:
test

Reply:
@sarneeh could be this helpful for creating the spy? 

https://github.com/jamiebuilds/ninos

Reply:
Yes, I agree with you the test case doesn't cover the actual behaviour.
However, I think it does still make sense to test if the cacheable-response returning `HIT` instead of `MISS` after empty 304 response because re-render did happen before empty 304 response.
We can assume that the server doesn't re-render when returning `HIT` because there is another test case `HIT for second access` which is using the same theory to test the lib.

Like you said, we may need to expand the test suite if we really want to test the actual server re-rendering.

Reply:
@iKoala That makes total sense! Let's merge this!